buildengine/engine.go

```diff
@@ -201,6 +206,56 @@ func (e *Engine) Build(ctx context.Context, modules ...string) error {
 	return nil
 }
 
+func (e *Engine) Deploy(ctx context.Context, replicas int32, waitForDeployOnline bool, modules ...string) error {
+	if len(modules) == 0 {
+		modules = maps.Keys(e.modules)
+	}
+
+	buildSubscription := e.builds.SubscribeSync(nil)
```

Comment:
I wouldn't bother with pubsub for an internal implementation. It's useful for decoupling components, but if all your code is in a single type then it's just overhead. I'd probably refactor the exist build code into a function that can accept a callback that is executed after each module is built.

Reply:
Oh I like that! I'll give it a go